kindlyops/mappamundi
# WIP moving from Auth0 to Keycloak.

## Description
Changing from Auth0 to Keycloak for authentication.

## TODO / Discussion list

- [x] proxy /auth to keycloak server in dev environment with a single realm
- [x] load keycloak.js from keycloak server
- [x] get logout working
- [x] Get initial login to send the elm message to re-render the page
- [x] update README to replace instructions for Auth0 with keycloak
- [x] automate the setup of keycloak realm and a test user in dev environment
- [x] adjust the data structures to fit the new profile object from keycloak
- [x] get links to user profile editing working.

I'm explicitly leaving styling/theming of keycloak for a later branch. There will be much more that we can do in later branches with mandatory flows, groups -> orgs, and attribute mapping into default claims, but I'm trying to have this branch be sufficient for us to have a semi working user profile and JWT that we can use in the UI and in REST API calls.

## How to test

* Load the webui main page. You should be redirected to keycloak login page
* Login with test credentials
* You should see the logged in home page
* Logout using the logout button in Elm, you should be logged out and sent back to the login screen
* login with test credentials again
* visit the Keycloak admin console in another tab http://localhost:8080/auth/ and look at the sessions for the test user. Logout the session from 
## Checklist:


- [x] My pull request is based on the latest master branch
  `git pull --rebase origin master`.
- [x] My code is formatted with elm-format.
- [x] For visual changes, I've attached before and after screenshots
